File: stack/stack.go
package main

import "fmt"

type Stack []string

func (stack *Stack) push(item string) {
	*stack = append(*stack, item)
}

func (stack *Stack) pop() (popped string) {
	*stack, popped = (*stack)[:len(*stack)-1], (*stack)[len(*stack)-1]
	return
}

func (stack *Stack) peek() string {
	return (*stack)[len(*stack)-1]
}

func (stack *Stack) isEmpty() bool {
    return len(*stack) == 0
}

func (stack *Stack) size() int {
    return len(*stack)
}

func main() {
	s := make(Stack, 0)
	s.push("I")
    s.push("Am")
    s.push("batman")
	fmt.Printf("The stack looks like-\n%v\n", s)
    fmt.Printf("Popping the last element - %v\n", s.pop())
    fmt.Printf("Peeked at the top of the stack be like - %v\n", s.peek())
    fmt.Printf("Is the stack empty yet? - %v\n", s.isEmpty())
    fmt.Printf("The current size of the stack is - %v\n", s.size())
    s.push("BATMAN")
    fmt.Printf("Final stack after pushing looks like -\n%v\n", s)
}

File: stack/stack.py
class Stack(object):

    def __init__(self):
        self.items = []

    def __repr__(self):
        return str(self.items)

    def push(self, item):
        self.items.append(item)

    def pop(self):
        return self.items.pop()

    def peek(self):
        return self.items[-1]

    def is_empty(self):
        return len(self.items) == 0

    def size(self):
        return len(self.items)


def main():
    stack = Stack()
    stack.push("I")
    stack.push("Am")
    stack.push("Batman")
    print("This is the stack: {}".format(stack))
    print("The top item in the stack is {}".format(stack.peek()))
    print("Is the stack empty?: {}".format(stack.is_empty()))
    print("The stack holds {} items right now".format(stack.size()))
    print("Removing {} from the stack, and putting in Shinchan".
          format(stack.pop()))
    stack.push("Shinchan")
    print("The final stack is {}".format(stack))


if __name__ == '__main__':
    main()
